### DISCUSSION TITLE
How can I Port forward in GitHub action VM, using ngrok right now for it

### DISCUSSION THREAD

**Discussion by kcubeterm** [community/community#26088] [Actions]
<p>Is there any way exist which can help me to port forwarding ,</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/kcubeterm">@kcubeterm</a>,</p>
<p>Glad to see you in Github Community Forum!</p>
<p>Please check my workflow <a href="https://github.com/weide-zhou/ticket13/actions/runs/211772726/workflow" rel="nofollow noopener">file</a> for port forward with ngrok. Confirmed it can be accessed via ssh to the port on local machine.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/b/bb3b1b25d6e4a09069e044ba5a60a282b4dcb24e.png" data-download-href="/uploads/short-url/qIjV6Grf9mdOmjD4b8nAmYcOnjE.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181095164-2d30e8f5-a137-447e-b8d0-dd6314a3330b.png" alt="image" data-base62-sha1="qIjV6Grf9mdOmjD4b8nAmYcOnjE" width="690" height="135" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/b/bb3b1b25d6e4a09069e044ba5a60a282b4dcb24e_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1102×216 11.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Thanks</p>

**Comment by kcubeterm** [comment]
<p>Thanks, yes and I am using ngrok right now I meant , can we port forwarding with ufw .</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/kcubeterm">@kcubeterm</a>,</p>
<p>Thanks for your reply!</p>
<p>You can create a self-hosted runner with local ubuntu machine, and create port forwarding with ufw on it. can refer to doc <a href="https://gist.github.com/kimus/9315140" rel="nofollow noopener">here</a>.</p>
